transmission_interface/test/differential_transmission_test.cpp

```diff
@@ -0,0 +1,457 @@
+    auto joint2_handle = JointHandle("joint2", HW_IF_VELOCITY, j_vec[1]);
+    trans.configure({joint1_handle, joint2_handle}, {a1_handle, a2_handle});
+    trans.actuator_to_joint();
+    EXPECT_NEAR(0.0, j_val[0], EPS);
+    EXPECT_NEAR(0.5, j_val[1], EPS);
+  }
+
+  // Position interface
+  {
+    auto a1_handle = ActuatorHandle("act1", HW_IF_POSITION, a_vec[0]);
+    auto a2_handle = ActuatorHandle("act2", HW_IF_POSITION, a_vec[1]);
+    auto joint1_handle = JointHandle("joint1", HW_IF_POSITION, j_vec[0]);
+    auto joint2_handle = JointHandle("joint2", HW_IF_POSITION, j_vec[1]);
+    trans.configure({joint1_handle, joint2_handle}, {a1_handle, a2_handle});
+    trans.actuator_to_joint();
+    EXPECT_NEAR(0.0, j_val[0], EPS);
+    EXPECT_NEAR(0.5, j_val[1], EPS);
+  }
+}
+
+TEST_F(WhiteBoxTest, MoveBothJoints)
+{
+  // NOTE: We only test the actuator->joint map,
+  // as the joint->actuator map is indirectly validated in the test that
+  // checks that actuator->joint->actuator == identity.
+
+  std::vector<double> actuator_reduction = {10.0, -20.0};
+  std::vector<double> joint_reduction = {-2.0, 4.0};
+  std::vector<double> joint_offset = {-2.0, 4.0};
+
+  DifferentialTransmission trans(actuator_reduction, joint_reduction, joint_offset);
+
+  // Actuator input (used for effort, velocity and position)
+  *a_vec[0] = 3.0;
+  *a_vec[1] = 5.0;
+
+  // Effort interface
+  {
+    auto a1_handle = ActuatorHandle("act1", HW_IF_EFFORT, a_vec[0]);
+    auto a2_handle = ActuatorHandle("act2", HW_IF_EFFORT, a_vec[1]);
+    auto joint1_handle = JointHandle("joint1", HW_IF_EFFORT, j_vec[0]);
+    auto joint2_handle = JointHandle("joint2", HW_IF_EFFORT, j_vec[1]);
+    trans.configure({joint1_handle, joint2_handle}, {a1_handle, a2_handle});
+    trans.actuator_to_joint();
+    EXPECT_NEAR(140.0, j_val[0], EPS);
+    EXPECT_NEAR(520.0, j_val[1], EPS);
+  }
+
+  // Velocity interface
+  {
+    auto a1_handle = ActuatorHandle("act1", HW_IF_VELOCITY, a_vec[0]);
+    auto a2_handle = ActuatorHandle("act2", HW_IF_VELOCITY, a_vec[1]);
+    auto joint1_handle = JointHandle("joint1", HW_IF_VELOCITY, j_vec[0]);
+    auto joint2_handle = JointHandle("joint2", HW_IF_VELOCITY, j_vec[1]);
+    trans.configure({joint1_handle, joint2_handle}, {a1_handle, a2_handle});
+    trans.actuator_to_joint();
+    EXPECT_NEAR(-0.01250, j_val[0], EPS);
+    EXPECT_NEAR(0.06875, j_val[1], EPS);
+  }
+
+  // Position interface
+  {
+    auto a1_handle = ActuatorHandle("act1", HW_IF_POSITION, a_vec[0]);
+    auto a2_handle = ActuatorHandle("act2", HW_IF_POSITION, a_vec[1]);
+    auto joint1_handle = JointHandle("joint1", HW_IF_POSITION, j_vec[0]);
+    auto joint2_handle = JointHandle("joint2", HW_IF_POSITION, j_vec[1]);
+    trans.configure({joint1_handle, joint2_handle}, {a1_handle, a2_handle});
+    trans.actuator_to_joint();
+    EXPECT_NEAR(-2.01250, j_val[0], EPS);
+    EXPECT_NEAR(4.06875, j_val[1], EPS);
+  }
+}
+
+int main(int argc, char ** argv)
```

Comment:
Why main? Usually, we do not have it...

Reply:
Good catch! I've removed it from here and also from simple_transmission. I also cleaned up the cmake a bit, no need for these tests to depend on transmission parser.